### PR TITLE
fix: state cache falls through to slow path on error

### PR DESCRIPTION
**Pull request by iand**
Prevents cache failures causing extraction to fail

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,54 +0,0 @@
-# This file was generated with `make docker-files` and should not
```
Comment (frrist) [inline Dockerfile:1]: You're probably aware - undo this change.

File: Dockerfile.dev [hunk 2]
```diff
@@ -1,42 +0,0 @@
-# This file was generated with `make docker-files` and should not
```
Comment (frrist) [inline Dockerfile.dev:1]: ditto

File: lens/util/store.go [hunk 3]
```diff
@@ -307,6 +302,20 @@ func (cas *CachingStateStore) Get(ctx context.Context, c cid.Cid, out interface{
...
 }
 
+func (cas *CachingStateStore) tryAssign(value interface{}, out interface{}) error {
+	o := reflect.ValueOf(out).Elem()
+	if !o.CanSet() {
+		return xerrors.Errorf("out parameter cannot be set")
+	}
+
+	if !value.(reflect.Value).Type().AssignableTo(o.Type()) {
+		return xerrors.Errorf("out parameter cannot be assigned cached value")
```
Comment (frrist) [inline lens/util/store.go:312]: include the types in the error message

File: lens/util/store.go [hunk 4]
```diff
@@ -307,6 +302,20 @@ func (cas *CachingStateStore) Get(ctx context.Context, c cid.Cid, out interface{
 	return nil
 }
 
+func (cas *CachingStateStore) tryAssign(value interface{}, out interface{}) error {
+	o := reflect.ValueOf(out).Elem()
+	if !o.CanSet() {
+		return xerrors.Errorf("out parameter cannot be set")
```
Comment (frrist) [inline lens/util/store.go:308]: include type in error message

**Review by frrist — APPROVED** [review]
Nice! This is ready to merge once comments are addressed.

File: Dockerfile (hunk 1 above)
Comment (iand) [reply Dockerfile:1]: Hmm. make clean deletes this file so what is the expected pattern for building? Do I need to regerenate the dockerfiles and recommit every time?

File: lens/util/store.go (hunk 3 above)
Comment (iand) [reply lens/util/store.go]: done

File: lens/util/store.go (hunk 4 above)
Comment (iand) [reply lens/util/store.go]: done

**Comment by iand** [comment]
@frrist restored dockerfiles

**Comment by frrist** [comment]
Looks good, my approval is sticky.
